src/main/java/com/metamx/common/parsers/TimestampParser.java

```diff
@@ -28,76 +28,107 @@
 {
   private static final Logger log = new Logger(TimestampParser.class);
 
-  public static Function<String, DateTime> createTimestampParser(
+  public static Function<Object, DateTime> createTimestampParserGeneric(
```

Comment:
I wonder if we should leave the existing one as is, and create new one that is specific to Long values. That would probably cleaner, remove a lot of branching in this method, and also leave the signature more specific as to which types we expect.